server/build.gradle

```diff
@@ -173,6 +174,10 @@ tasks.withType(JavaCompile).configureEach {
   options.compilerArgs -= '-Xlint:unchecked'
 }
 
+compileJava {
```

Comment:
Sorry @Xtansia, to fill the gap in my understanding, adding `annotationProcessor` would not result into adding`-processor <...>` compiler argument out of the box? (I would have expected that to be fair)

Reply:
So `annotationProcessor` adds onto `-processorpath` as Gradle deviates from `javac` behaviour and doesn't allow searching compile classpath for processors. Which normally the processors would then be automatically discovered from via javac, however our extension of the java plugin automatically adds `-proc:none` unless we're explicitly specifying processors https://github.com/opensearch-project/OpenSearch/blob/74788c3f45895a90d030b2434539beac7a1b1723/buildSrc/src/main/java/org/opensearch/gradle/OpenSearchJavaPlugin.java#L167-L169

Reply:
You can see benchmarks needing to do the same thing here: https://github.com/opensearch-project/OpenSearch/blob/74788c3f45895a90d030b2434539beac7a1b1723/benchmarks/build.gradle#L49-L57

Reply:
Got it, thank you